### PR TITLE
Update scintilla for Qt6

### PR DESCRIPTION
**Pull request by frmdstryr**
Builds on #475. It seems like PyQt6 changes how enums values are exposed.  

One fix in q_icon_widget that may need addressed https://doc-snapshots.qt.io/qt6-dev/qlayout.html#getContentsMargins seems to have moved from QWidget.

### CODE REVIEW

File: enaml/qt/Qsci.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+if QT_API in PYSIDE2_API:
+    msg = 'the Qt Scintilla widget is only available when using PyQt'
+    raise ImportError(msg)
+
+if QT_API in PYQT6_API:
+    from PyQt6.Qsci import *
+elif QT_API in PYQT5_API:
+    from PyQt5.Qsci import *
+else:
```
Comment (MatthieuDartiailh) [inline enaml/qt/Qsci.py:18]: This dates from PyQt4 and can be removed. Also what is the situation with PySide6 ?

File: enaml/qt/docking/q_icon_widget.py [hunk 2]
```diff
@@ -93,6 +94,8 @@ def minimumSizeHint(self):
         size = self._icon_size
         if not size.isValid():
             size = QSize(16, 16)
+        if IS_QT6:
```
Comment (MatthieuDartiailh) [inline enaml/qt/docking/q_icon_widget.py:97]: Could you add a comment on this one ?

**Review by MatthieuDartiailh — CHANGES_REQUESTED** [review]
Since this was not caught in the CIs anything we could change to avoid future regressions ?

File: enaml/qt/docking/q_icon_widget.py (hunk 2 above)
Comment (frmdstryr) [reply enaml/qt/docking/q_icon_widget.py:97]: The function getContentsMargins is obsolete and https://doc.qt.io/qt-5/qwidget-obsolete.html 
it seems it has been removed.

File: enaml/qt/Qsci.py (hunk 1 above)
Comment (frmdstryr) [reply enaml/qt/Qsci.py:14]: I think pyside avoids it due to the GPL.

File: enaml/qt/docking/q_icon_widget.py (hunk 2 above)
Comment (frmdstryr) [reply enaml/qt/docking/q_icon_widget.py]: Updated to use `contentsMargins()` now

File: enaml/qt/docking/q_icon_widget.py [hunk 3]
```diff
@@ -93,8 +93,8 @@ def minimumSizeHint(self):
         size = self._icon_size
         if not size.isValid():
             size = QSize(16, 16)
-        left, top, right, bottom = self.getContentsMargins()
-        return size + QSize(left + right, top + bottom)
+        m = self.contentsMargins()
```
[the remark this answers is not in the document]
Comment (frmdstryr) [reply enaml/qt/docking/q_icon_widget.py:96]: A [search](https://github.com/nucleic/enaml/search?q=getContentsMargins) shows uses in `q_flow_layout.py` and `q_single_widget_layout.py` Both subclass QLayout which still has the method in Qt6.  Looks like you already did the label so I think that's all.

[the remark this answers is not in the document]
Comment (MatthieuDartiailh) [reply enaml/qt/docking/q_icon_widget.py:96]: Thanks for double checking

**Comment by MatthieuDartiailh** [comment]
Looks great, I am just bothered by the fact that running the example in CI did not catch any of those issues. How did you catch them ?

**Comment by frmdstryr** [comment]
These were found by running DeclaraCAD. 

It doesn't look like CI is running the live editor on Qt6?

**Comment by MatthieuDartiailh** [comment]
Indeed looks like I messed up the CI config. Could you replace the following snippet by the next one in ci.yml ?

```
 - name: Install QScintilla
        if: matrix.python-version == '3.9' && matrix.qt-binding == 'pyqt'
        run: |
          pip install Cython QScintilla
```

```
 - name: Install QScintilla (Qt5)
        if: matrix.python-version == '3.9' && matrix.qt-binding == 'pyqt' && matrix.qt-version== 5
        run: |
          pip install QScintilla
 - name: Install QScintilla (Qt6)
        if: matrix.python-version == '3.9' && matrix.qt-binding == 'pyqt' && matrix.qt-version== 6
        run: |
          pip install PyQt6-QScintilla
```

**Comment by MatthieuDartiailh** [comment]
Thanks. I will merge since otherwise the CIs won't kick in.

**Comment by frmdstryr** [comment]
Qt6 seems to run a bit smoother, at least on linux & wayland. Thanks for doing the updates!

**Pull request merged**